src/main/resources/update/gitbucket-core_4.24.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<changeSet>
+  <createTable tableName="ACCOUNT_EXTRA_MAIL_ADDRESS">
+    <column name="USER_NAME" type="varchar(100)" nullable="false"/>
+    <column name="EXTRA_MAIL_ADDRESS" type="varchar(100)" nullable="false"/>
+  </createTable>
+
+  <addPrimaryKey constraintName="IDX_ACCOUNT_EXTRA_MAIL_ADDRESS_PK" tableName="ACCOUNT_EXTRA_MAIL_ADDRESS" columnNames="USER_NAME, EXTRA_MAIL_ADDRESS"/>
+  <addUniqueConstraint constraintName="IDX_ACCOUNT_EXTRA_MAIL_ADDRESS_1" tableName="ACCOUNT_EXTRA_MAIL_ADDRESS" columnNames="EXTRA_MAIL_ADDRESS"/>
+</changeSet>
```

Comment:
I'm not sure that adding `ACCOUNT_EXTRA_MAIL_ADDRESS` table is the best solution. I think that we should add a table like `ACCOUNT_MAIL_ADDRESS` which has mail addresses (with unique constraint) of users and a flag to show which address is the primary address of the user.

However, in order to do it, we need large modification of our code base and database structure. So your solution is reasonable, I think. 👍

Reply:
Yes, I think single mail address table as better design for **new** applixation. But it requires hard migration, interface changes, etc. So, I selected this solution.

Reply:
Totally agree.